http/fetch/archive_fetcher.go

```diff
@@ -94,7 +218,7 @@ func (r *ArchiveFetcher) Fetch(archiveURL, digest, dir string) error {
 		archiveURL = u.String()
 	}
 
-	req, err := retryablehttp.NewRequest(http.MethodGet, archiveURL, nil)
+	req, err := retryablehttp.NewRequestWithContext(ctx, http.MethodGet, archiveURL, nil)
```

Comment:
I think we should preserve the previous behaviour, which is download first and proceed to file creation only if download is successful.

Reply:
fixed, thanks